diff --git a/google/cloud/bigtable/row.h b/google/cloud/bigtable/row.h
index a6eeba0..b34ef2b 100644
--- a/google/cloud/bigtable/row.h
+++ b/google/cloud/bigtable/row.h
@@ -32,18 +32,20 @@ inline namespace BIGTABLE_CLIENT_NS {
 class Row {
  public:
   /// Create a row from a list of cells.
-  Row(std::string row_key, std::vector<Cell> cells)
+  Row(absl::Cord row_key, std::vector<Cell> cells)
       : row_key_(std::move(row_key)), cells_(std::move(cells)) {}
+  Row(std::string row_key, std::vector<Cell> cells)
+      : Row(absl::Cord(std::move(row_key)), std::move(cells)) {}
 
   /// Return the row key. The returned value is not valid
   /// after this object is deleted.
-  std::string const& row_key() const { return row_key_; }
+  absl::Cord const& row_key() const { return row_key_; }
 
   /// Return all cells.
   std::vector<Cell> const& cells() const { return cells_; }
 
  private:
-  std::string row_key_;
+  absl::Cord row_key_;
   std::vector<Cell> cells_;
 };
 
