diff --git a/google/cloud/bigtable/tests/data_integration_test.cc b/google/cloud/bigtable/tests/data_integration_test.cc
index 8e92f83..457f700 100644
--- a/google/cloud/bigtable/tests/data_integration_test.cc
+++ b/google/cloud/bigtable/tests/data_integration_test.cc
@@ -88,9 +88,9 @@ void DataIntegrationTest::Apply(bigtable::Table& table, std::string row_key,
 void DataIntegrationTest::BulkApply(bigtable::Table& table,
                                     std::vector<bigtable::Cell> const& cells) {
   using namespace std::chrono;
-  std::map<std::string, bigtable::SingleRowMutation> mutations;
+  std::map<absl::Cord, bigtable::SingleRowMutation> mutations;
   for (auto const& cell : cells) {
-    std::string key = cell.row_key();
+    auto key = cell.row_key();
     auto inserted = mutations.emplace(key, bigtable::SingleRowMutation(key));
     inserted.first->second.emplace_back(bigtable::SetCell(
         cell.family_name(), cell.column_qualifier(),
